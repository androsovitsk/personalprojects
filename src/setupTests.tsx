import '@testing-library/jest-dom'
import React from 'react'
import { render } from '@testing-library/react'
import fetchMock from 'fetch-mock'
import TranslationProvider from './ui/components/TranslationProvider/TranslationProvider'
import LanguageTypes from './ui/components/TranslationProvider/types/LanguageTypes'
import resourceBundles from './locale'
import { createTheme, ThemeProvider } from '@mui/material'

beforeEach(() => {
  fetchMock.reset()
  jest.clearAllMocks()
})

const Wrapper: React.FC<React.PropsWithChildren> = ({ children }) => {
  return (
    <ThemeProvider theme={createTheme({})}>
      <TranslationProvider
        defaultLanguage={LanguageTypes.ENGLISH}
        resourceBundles={resourceBundles}
      >
        {children}
      </TranslationProvider>
    </ThemeProvider>
  )
}

export const renderWithProviders = (element: React.ReactElement) =>
  render(element, {
    wrapper: Wrapper
  })
