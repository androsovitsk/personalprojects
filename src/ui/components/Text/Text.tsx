import React from 'react'
import Typography from '@mui/material/Typography'
import { TypographyProps } from '@mui/material/Typography'

interface ITextProps extends TypographyProps {
  noMargin?: boolean
}
const Text: React.FC<ITextProps> = ({ noMargin, children, ...props }) => {
  return (
    <Typography {...(noMargin && { sx: { marginBottom: 0 } })} {...props}>
      {children}
    </Typography>
  )
}

Text.displayName = 'Text'

export default Text
