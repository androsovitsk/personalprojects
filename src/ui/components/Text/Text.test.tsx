import { renderWithProviders } from '../../../setupTests'
import Text from './Text'

const testText = 'Test text'

describe('Text', () => {
  it('should match the snapshot', () => {
    const view = renderWithProviders(<Text>{testText}</Text>)

    expect(view.asFragment()).toMatchSnapshot()
  })

  it('should match the snapshot when no margin is applied', () => {
    const view = renderWithProviders(<Text noMargin>{testText}</Text>)

    expect(view.asFragment()).toMatchSnapshot()
  })
})
