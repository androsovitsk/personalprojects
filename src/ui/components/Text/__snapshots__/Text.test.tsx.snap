// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should match the snapshot 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Test text
  </p>
</DocumentFragment>
`;

exports[`Text should match the snapshot when no margin is applied 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1wczmbf-MuiTypography-root"
  >
    Test text
  </p>
</DocumentFragment>
`;
